history/src/block_stores/inmemory_block_store.rs

```diff
@@ -52,12 +53,18 @@ impl InmemoryBlockStore {
 
 #[async_trait]
 impl BlockStorageInterface for InmemoryBlockStore {
-    async fn save(&self, block: ProducedBlock) {
+    async fn save(&self, block: ProducedBlock) -> anyhow::Result<()> {
```

Comment:
we should prevent a slot going back in the states model

again: IMO the modeling is not reflecting the state model of slot->commitment_status correctly